src/pages/index.js

```diff
@@ -31,33 +30,23 @@ const config = { spring, easings, getMargins: true }
 
 const IndexPage = () => {
   const morph = useMorph(config)
-  const [renderSplash, setRenderSplash] = useState(true)
-
-  useEffect(() => {
-    disablePageScroll(document.documentElement)
-
-    setTimeout(() => setRenderSplash(false), 750)
-    setTimeout(() => enablePageScroll(document.documentElement), 1000)
-  }, [])
+  const [hidingTittlePlanet, setHidingTittlePlanet] = useState(true)
 
   return (
-    <Layout>
-      <noscript>
-        <style>
-          {`
-            #splash-screen{
-              display: none
-            }
-          `}
-        </style>
-      </noscript>
-
-      <SEO />
-      <Splash show={renderSplash} morph={morph} />
-      <HeroSection planetMorph={morph} hidePlanet={renderSplash} />
-      <VenturesSection />
-      <AboutSection />
-    </Layout>
+    <>
+      <SplashScreen
+        lockScrollFor={1000}
+        morph={morph}
+        onHide={() => setHidingTittlePlanet(false)}
```

Comment:
Advise against inline arrow functions, regardless of what Gabriel might say :joy:

Reply:
Do you have any reason for that advice or is it subjective?

Both ways work for me actually. Using a variable allows the component usage to be more semantic and readable. On the other hand functions simple as this one are fairly easy to read and the code remains simpler and more succint.

Changed anyhow :+1:

Reply:
Basically it's changing the reference every time the parent re-renders. If the parent re-renders that inline expression is evaluated and changes it's reference, causing a re-render of the `SplashScreen` component, even if the other props don't change. By defining the function early, you maintain that reference and avoid extra renders.

Reply:
But performance impacts are negligible, usually, the Pong game we made with React would suffer tremendously with this but we ported everything to `requestAnimationFrame`  anyhow so there were almost no React re-renders.

But basically it's your first option. I like defining variables with functions, makes stuff more readable. I write components usually with `hooks - variables - handlers - jsx`. So people always expect extra behavior in the first section of the component, and the `jsx` is just the structure.